### FTDS - 015
### Rio R

In [1]:
import pandas as pd
import numpy as np

#### Loading Dataset

In [3]:
data = pd.read_csv('https://raw.githubusercontent.com/rafifaditio/Dataset/main/BMR.csv')
data

Age  Gender  Height (cm)  Weight (kg)  Index
0     21    Male          162           58      2
1     44  Female          188           80      2
2     45    Male          177           61      2
3     50    Male          192          101      3
4     28  Female          195           61      1
..   ...     ...          ...          ...    ...
145   33    Male          142           71      4
146   32  Female          162           58      2
147   29    Male          174           54      1
148   33  Female          197           72      2
149   23    Male          198           50      0

[150 rows x 5 columns]

In [51]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 6 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Age          150 non-null    int64  
 1   Gender       150 non-null    object 
 2   Height (cm)  150 non-null    int64  
 3   Weight (kg)  150 non-null    int64  
 4   Index        150 non-null    int64  
 5   BMR          150 non-null    float64
dtypes: float64(1), int64(4), object(1)
memory usage: 7.2+ KB


- A. Buatlah dua fungsi eksplisit masing-masing memuat perhitungan BMR laki-laki dan BMR Perempuan yang nantinya akan digunakan untuk perhitungan

In [41]:
def BMR_Male (weight,height,age):
  calc = (66 + (137*weight) + (5*height) - (6.8*age))
  return calc

In [42]:
BMR_Male(58,162,21)

8679.2

In [43]:
def BMR_Female (weight,height,age):
  calc = (655 + (9.6*weight) + (1.8*height) - (4.7*age))
  return calc

In [44]:
BMR_Female (80,188,44)

1554.6000000000001

- B. Buat kolom baru dengan nama kolom BMR berisikan hasil perhitungan kebutuhan energi dasar (BMR) harian sesuai dengan jenis kelamin masing-masing pelanggan. (Hint: Gunakan for loop dan if statement untuk mengerjakan perintah nomor b, kamu bisa menggunakan list untuk menyimpan perhitungan sementara, setelahnya dapat dimasukkan ke kolom BMR).

In [45]:
data1.head()

Age  Gender  Height (cm)  Weight (kg)  Index      BMR
0   21    Male          162           58      2   8708.2
1   44  Female          188           80      2   1554.6
2   45    Male          177           61      2   9032.5
3   50    Male          192          101      3  14573.5
4   28  Female          195           61      1   1460.0

In [46]:
data1.shape[0]

150

In [47]:
bmr = []

for i in range (0,data1.shape[0]) :
  if data1.loc[i,'Gender'] == 'Male' :
    calc = BMR_Male(weight=data1.loc[i,'Weight (kg)'], 
                    height=data1.loc[i,'Height (cm)'], 
                    age=data1.loc[i,'Age'])
    bmr.append(calc)
  elif data1.loc[i,'Gender'] == 'Female' :
    calc = BMR_Female (weight=data1.loc[i,'Weight (kg)'],
                       height=data1.loc[i,'Height (cm)'],
                       age=data1.loc[i,'Age'])
    bmr.append(calc)


In [48]:
data1['BMR']= bmr
data1

Age  Gender  Height (cm)  Weight (kg)  Index      BMR
0     21    Male          162           58      2   8679.2
1     44  Female          188           80      2   1554.6
2     45    Male          177           61      2   9002.0
3     50    Male          192          101      3  14523.0
4     28  Female          195           61      1   1460.0
..   ...     ...          ...          ...    ...      ...
145   33    Male          142           71      4  10278.6
146   32  Female          162           58      2   1353.0
147   29    Male          174           54      1   8136.8
148   33  Female          197           72      2   1545.7
149   23    Male          198           50      0   7749.6

[150 rows x 6 columns]

- C. Hitunglah rata-rata berat badan untuk Jenis kelamin laki-laki dengan umur lebih dari sama dengan 24 tahun hingga kurang dari sama dengan 35 tahun.

In [49]:
# Copy dataset terlebih dahulu
data1 = data
data1

Age  Gender  Height (cm)  Weight (kg)  Index      BMR
0     21    Male          162           58      2   8679.2
1     44  Female          188           80      2   1554.6
2     45    Male          177           61      2   9002.0
3     50    Male          192          101      3  14523.0
4     28  Female          195           61      1   1460.0
..   ...     ...          ...          ...    ...      ...
145   33    Male          142           71      4  10278.6
146   32  Female          162           58      2   1353.0
147   29    Male          174           54      1   8136.8
148   33  Female          197           72      2   1545.7
149   23    Male          198           50      0   7749.6

[150 rows x 6 columns]

In [50]:
data1[(data1['Gender']=='Male') &
      (data1['Age']>=24) &
      (data1['Age']<=35)]['Weight (kg)'].mean()

86.41176470588235

- Rata - rata berat badan laki laki dengan umur lebih dari sama dengan 24 tahun hingga kurang dari sama dengan 35 tahun adalah 86.4 kg.